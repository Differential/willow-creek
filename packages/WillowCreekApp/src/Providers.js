import { nest } from 'recompose';
import { Providers } from '@apollosproject/ui-kit';
import ClientProvider from './client';

export default nest(ClientProvider, Providers);
