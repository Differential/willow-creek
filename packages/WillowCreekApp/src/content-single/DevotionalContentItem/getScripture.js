import gql from 'graphql-tag';

export default gql`
  query getScripture($itemId: ID!) {
    node(id: $itemId) {
      __typename
      id
      ... on DevotionalContentItem {
        scriptures {
          id
          html
          reference
        }
      }
    }
  }
`;
