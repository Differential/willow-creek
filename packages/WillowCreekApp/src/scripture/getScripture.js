import gql from 'graphql-tag';

export default gql`
  query GetScripture($query: String!) {
    scripture(query: $query) {
      reference
      html
      copyright
    }
  }
`;
