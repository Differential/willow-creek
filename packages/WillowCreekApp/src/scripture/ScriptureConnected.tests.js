import React from 'react';
import renderer from 'react-test-renderer';
import wait from 'waait';

import Providers from 'WillowCreekApp/src/Providers';

import getScripture from './getScripture';
import ScriptureConnected from '.';

const mocks = {
  request: {
    query: getScripture,
    variables: { query: 'Genesis 1:1' },
  },
  result: {
    data: {
      scripture: {
        id: 'GEN.1.1',
        reference: 'Genesis 1:1',
        copyright: 'PUBLIC DOMAIN',
        html:
          '<p class="p"><span data-number="1" class="v">1</span>In the beginning, God created the heavens and the earth. </p>',
      },
    },
  },
};

describe('ScriptureConnected component', () => {
  it('renders without errors', async () => {
    const tree = renderer.create(
      <Providers mocks={[mocks]} addTypename={false}>
        <ScriptureConnected references={['Genesis 1:1']} />
      </Providers>
    );
    await wait(0); // wait for response from graphql
    expect(tree).toMatchSnapshot();
  });
});
