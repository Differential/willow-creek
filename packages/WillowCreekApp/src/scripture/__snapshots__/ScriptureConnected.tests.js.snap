// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ScriptureConnected component renders without errors 1`] = `
<RCTSafeAreaView
  onLayout={[Function]}
  style={
    Object {
      "bottom": 0,
      "left": 0,
      "position": "absolute",
      "right": 0,
      "top": 0,
    }
  }
>
  <View
    onLayout={[Function]}
    style={
      Object {
        "flex": 1,
      }
    }
  />
  <View
    style={
      Object {
        "bottom": 0,
        "left": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
      }
    }
  >
    <View>
      <Text
        style={
          Object {
            "color": "#303030",
            "fontFamily": "InterUI-Bold",
            "fontSize": 16,
            "lineHeight": 23.04,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#303030",
              "fontFamily": "InterUI-Bold",
              "fontSize": 16,
              "lineHeight": 23.04,
            }
          }
        >
          Genesis 1:1
        </Text>
         
        <Text
          style={
            Object {
              "color": "#505050",
              "fontFamily": "InterUI-Bold",
              "fontSize": 12,
              "lineHeight": 17.28,
            }
          }
        >
          WEB
        </Text>
      </Text>
      <View
        lineNumber={8}
        onReady={true}
        style={
          Object {
            "paddingBottom": 17.28,
          }
        }
      >
        <View>
          <View
            style={
              Object {
                "paddingBottom": 17.28,
              }
            }
          >
            <Text
              bold={false}
              italic={false}
              style={
                Object {
                  "color": "#303030",
                  "fontFamily": "DroidSerif",
                  "fontSize": 16,
                  "lineHeight": 26,
                }
              }
            >
              <Text
                bold={false}
                italic={false}
                style={
                  Object {
                    "color": "#505050",
                    "fontFamily": "DroidSerif",
                    "fontSize": 9.6,
                    "lineHeight": 26,
                  }
                }
              >
                 
                <Text>
                  1
                </Text>
                Â 
              </Text>
              <Text
                bold={false}
                italic={false}
                style={
                  Object {
                    "color": "#303030",
                    "fontFamily": "DroidSerif",
                    "fontSize": 16,
                    "lineHeight": 26,
                  }
                }
              >
                <Text>
                  In the beginning, God created the heavens and the earth. 
                </Text>
              </Text>
            </Text>
          </View>
        </View>
      </View>
      <Text
        bold={false}
        italic={false}
        style={
          Object {
            "color": "#505050",
            "fontFamily": "InterUI-Regular",
            "fontSize": 9.6,
            "lineHeight": 10.37,
          }
        }
      >
        
      </Text>
    </View>
  </View>
</RCTSafeAreaView>
`;
